medusa/purge.py

```diff
@@ -140,9 +140,10 @@ def cleanup_obsolete_files(storage, fqdn):
     for path in paths_in_storage - paths_in_manifest:
         logging.debug("  - [{}] exists in storage, but not in manifest".format(path))
         obj = storage.storage_driver.get_blob(path)
-        nb_objects_purged += 1
-        total_purged_size += int(obj.size)
-        storage.storage_driver.delete_object(obj)
+        if obj is not None:
```

Comment:
In which case could we be getting a `None` object? 
The `path` comes from objects listed from storage, which means we should get something in all cases.